plutus-core/untyped-plutus-core/test/Transform/Simplify.hs

```diff
@@ -226,6 +226,35 @@ multiApp = runQuote $ do
       app = mkIterAppNoAnn lam [mkConstant @Integer () 1, mkConstant @Integer () 2, mkConstant @Integer () 3]
   pure app
 
+-- | This is the first example in Note [CSE].
+cse1 :: Term Name PLC.DefaultUni PLC.DefaultFun ()
```

Comment:
I feel like we've discussed quite a number of examples so far - this should have more than two tests!

Reply:
I'll add the third example in the Note, as well as the pathological example.